streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -704,7 +704,7 @@
                     CommonClientConfigs.RECONNECT_BACKOFF_MAX_MS_DOC)
             .define(RETRIES_CONFIG,
                     Type.INT,
-                    0,
+                    5,
```

Comment:
@guozhangwang @vvcephei Not sure if we need a KIP for changing this default value? For the global thread, increasing the retires is just an "improvement". For the `TimeoutException` that we don't handle at all yet, there is no backward compatibility concern.